src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -246,4 +248,38 @@ public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([F
         var responses = secrets.Select(s => new BaseSecretResponseModel(s));
         return new ListResponseModel<BaseSecretResponseModel>(responses);
     }
+
+    [HttpPost("organizations/{organizationId}/secrets/move")]
+    public async Task BulkMoveToProjectAsync(Guid organizationId, [FromBody] BulkMoveToProjectsRequestModel request)
+    {
+        var secrets = (await _secretRepository.GetManyByIds(request.Secrets)).ToList();
+        if (!secrets.Any() || secrets.Count != request.Secrets.Count)
+        {
+            throw new NotFoundException();
+        }
+
+        if (secrets.Any(secret => secret.OrganizationId != organizationId) ||
+            !_currentContext.AccessSecretsManager(organizationId))
+        {
+            throw new NotFoundException();
+        }
+
+        var authorizationResult = await _authorizationService.AuthorizeAsync(
+            User, secrets.AsReadOnly(), BulkSecretOperations.Update);
+
+        if (!authorizationResult.Succeeded)
+        {
+            throw new NotFoundException();
+        }
+
+        var project = await _projectRepository.GetByIdAsync(request.Project);
```

Comment:
Can we move this project fetch up with the secrets fetching so we can check that both exist and both are in the same organization first?

Reply:
I moved the project fetch and validation before we do anything with secrets this way we only get a single item from the database if it's wrong and fail the request vs getting `n` number of secrets and validating those before getting to the project. Should make the load on our DB be lower in the worst case, is that what you meant by this or did I misunderstand?